### ISSUE TITLE
[Enhancement] Offer to help fix Windows issues

### ISSUE THREAD

**Issue by ericgrosse** [kmagiera/babel-watch#19]
It states in your documentation:

_(It doesn't work on Windows as it uses unix named pipes, if you want to help with getting that fixed, feel free to contact me)_

I'd be willing to help fix this issue :)

**Comment by laggingreflex** [comment]
+1 This would be awesome. @kmagiera please consider adding windows support.

**Comment by kmagiera** [comment]
Hey @ericgrosse sorry for late response. I'd be happy to accept PRs for windows support. If you need some help or guidance please message me on twitter or we can continue the discussion here if you prefer (I promise to respond timely the next time)

**Comment by cristian-eriomenco** [comment]
@kmagiera As far as I understand its because the mkfifo command, there is no native implementation for this on windows? So it should be a matter of using either another method of accomplishing whatever mkfifo was used for or a 3-rd party library?

**Comment by kmagiera** [comment]
@cristian-eriomenco yes, that's more or less correct. So on top of what you just said the effect of `mkfifo` command is utilised. So the command creates a "named pipe", which is an object in the filesystem. Then that object is used to read and write to it using node's file system API (https://nodejs.org/api/fs.html) – and this is how main process and the app process communicate with each other. The trick here is that it's important that such an object is accessible using `fs` API and that is because we need to be able to "load" transpiled JS files content **synchronously**. It comes from the fact that `require` method is **synchronous**, which means when it returns, node gets a content of the loaded file. This cannot be done using concepts such as sockets, as there is no way to synchronously read from the socket and therefore we wouldn't be able to read the content before the method need to return.

If there is no alternative that would allow us for utilising standard node APIs (such as `fs`) I'm afraid the only option would be to build a native (c++ or whatever) addon – then with such an addon we can implement synchronous socket reader (or implement some other synchronous communication mechanism). On the other hand that would complicate module distribution (compiling after installation), so I prefer to avoid that if possible

**Comment by laggingreflex** [comment]
Why is `mkfifo` necessary? Can't it be a simple file? I tested on Windows by [replacing `mkfifo`](https://github.com/kmagiera/babel-watch/blob/22119922c65edc4f6e8cebe16ac382ac4582f191/babel-watch.js#L233) with simply `echo. > ${filename}` (which is the Windows equivalent of `touch filename` in linux)

```
// execSync(`mkfifo -m 0666 ${pipeFilename}`);
execSync(`echo. > ${pipeFilename}`);
```

It seems to work (in simple testing at least, not sure if there might be any caveats with this)

![scr](https://cloud.githubusercontent.com/assets/2523987/18552022/6f385964-7b78-11e6-83fa-a7efbf612821.png)

**Comment by kmagiera** [comment]
@laggingreflex the problem with just writing to a file is that all the things you've written are going to be flushed at some point to the disk and the file will grow over time. 

I've found this answer on stackoverflow http://stackoverflow.com/a/32172145/1665044 where named pipes are accessed on nodejs but using socket API. Unfortunately I don't have an easy way to test that out on windows. If that could also work with filesystem API maybe we can leverage that? I don't understand if this code is responsible for creating the pipe though (but at least that's what the question is all about) so I'd guess that maybe it's not necessary on windows?

**Comment by laggingreflex** [comment]
I noticed [**fifoJS**](https://github.com/zacgeis/fifoJS) on that stackoverflow link, could that work? Although right now it doesn't install for me on Windows, but maybe it can be supported. I also came across [**mkfifo**](https://github.com/avz/node-mkfifo) which didn't install either.

**Comment by kmagiera** [comment]
Thanks @laggingreflex for your pull request. Your changes are now merged and published as a part of 2.0.3-rc0 
I'm closing this issue for now. If anyone is experiencing some problems on windows please feel free to reopen this issue or to create a new one
